pkg/engine/internal/imageverifier.go

```diff
@@ -467,6 +473,15 @@ func (iv *ImageVerifier) buildCosignVerifier(
 		opts.Roots = imageVerify.Roots
 	}
 
+	if len(imageVerify.ImageRegistryCredentials.Secrets) != 0 {
```

Comment:
Why do we need this ?

Reply:
in the case where, the flag is not set in the args but imageRegistryCredentials is set in the policy